misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/ObjectMock.java

```diff
@@ -245,6 +365,158 @@ public static BError validateArguments(BObject caseObj) {
+                    } else if (!TypeChecker.checkIsType(it.next(), paramType)) {
+                        return ErrorCreator.createError(
+                                MockConstants.TEST_PACKAGE_ID,
+                                MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                                StringUtils.fromString(detail),
+                                null,
+                                new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+                    }
+                }
+                break;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Validates the resource function arguments.
+     *
+     * @param caseObj ballerina object that contains information about the case to register
+     * @return an optional error if a validation fails
+     */
+    public static BError validateResourceArguments(BObject caseObj) {
+        GenericMockObjectValue genericMock = (GenericMockObjectValue) caseObj.getObjectValue(
+                StringUtils.fromString(MockConstants.MOCK_OBJECT));
+        String functionName = caseObj.getStringValue(StringUtils.fromString(MockConstants.FUNCTION_NAME)).toString();
+        BArray argsList = caseObj.getArrayValue(StringUtils.fromString(MockConstants.ARGS));
+        String functionPattern = getFunctionNameForResourcePath(functionName);
+        String accessor = caseObj.getStringValue(StringUtils.fromString(MockConstants.ACCESSOR)).toString();
+        functionPattern = MockConstants.DOLLAR_RESOURCE_SEPARATOR + accessor +
+                MockConstants.DOLLAR_RESOURCE_SEPARATOR + functionPattern;
+        for (ResourceMethodType attachedFunction : ((BClientType) genericMock.getType()).getResourceMethods()) {
+            if (attachedFunction.getName().endsWith(functionPattern)) {
+                int pathSegmentCount = (int) functionPattern.chars().filter(ch -> ch ==
+                        MockConstants.PATH_PARAM_PLACEHOLDER.charAt(0)).count() -
+                        (functionPattern.endsWith(MockConstants.REST_PARAM_PLACEHOLDER) ? 1 : 0);
+                // validate the number of arguments provided
+                if (argsList.size() > attachedFunction.getType().getParameters().length - pathSegmentCount) {
+                    String detail = "too many argument provided to mock the function '" + functionName + "()'";
+                    return ErrorCreator.createError(
+                            MockConstants.TEST_PACKAGE_ID,
+                            MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                            StringUtils.fromString(detail),
+                            null,
+                            new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+                }
+
+                // validate if each argument is compatible with the type given in the function signature
+                int i = 0;
+                for (BIterator it = argsList.getIterator(); it.hasNext(); i++) {
```

Comment:
can we use the stream API to iterate the arg list?

Reply:
There is no direct way to use the stream API with `BArray`.  Methods which are needed to extract the Java array from `BArray` can vary depending on the type stored. (Eg.. If we have `String` then we need to use `getStringArray`, for `int` we need to use `getIntArray`). Accessing through BIterator is uniform for all types and easier here.